File: app/src/main/java/kr/co/company/capstone/service/NotificationService.java
package kr.co.company.capstone.service;

import kr.co.company.capstone.dto.notification.NotificationInfoListResponse;
import kr.co.company.capstone.dto.notification.NotificationDetailListResponse;
import kr.co.company.capstone.dto.notification.NotificationDetailResponse;
import kr.co.company.capstone.dto.notification.NotificationInfoResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface NotificationService {

    // 단건 알림 조회 (알림 하나 클릭 했을 때)
    @GET("/notifications/{notificationId}")
    Call<NotificationDetailResponse> findNotification(@Path("notificationId") long notificationId);

    // 유저별 알림 조회
    @GET("/notifications")
    Call<NotificationInfoListResponse<NotificationInfoResponse>> notifications(@Query("cursorId") Long cursorId, @Query("size") int size);

    // 목표 수행 완료 알림 조회
    @GET("/notifications/goal-complete")
    Call<NotificationDetailListResponse> findGoalCompleteNotifications();

    static NotificationService getService(){
        return RetrofitBuilder.getRetrofit().create(NotificationService.class);
    }
}

File: app/src/main/java/kr/co/company/capstone/service/NotificationService.kt
package kr.co.company.capstone.service

import kr.co.company.capstone.dto.notification.NotificationDetailResponse
import kr.co.company.capstone.dto.notification.NotificationResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface NotificationService {
    // 단건 알림 조회 (알림 하나 클릭 했을 때)
    @GET("/notifications/{notificationId}")
    fun loadSpecificNotificationInformation(@Path("notificationId") notificationId: Long): Call<NotificationDetailResponse>

    // 알림 조회
    @GET("/notifications")
    fun loadNotifications(): Call<NotificationResponse>

    // 추가 알림 조회
    @GET("/notifications")
    fun loadAdditionalNotifications(
        @Query("cursorId") cursorId : Long,
        @Query("size") size : Int
    ): Call<NotificationResponse>

    companion object {
        fun service(): NotificationService{
            return RetrofitBuilder.getRetrofit().create(NotificationService::class.java)
        }
    }
}
